src/utils/server-request.js

```diff
@@ -0,0 +1,243 @@
+import axios from "axios";
+import { isAlreadyAdded } from "./array-update-functions";
+
+export const getProductsFromServer = async (url) => {
+  const res = await axios({ method: "GET", url: url });
+
+  if (res.status === 200 || res.status === 201) {
+    return res;
+  } else {
+    throw new Error("Failed to get products");
+  }
+};
+
+export const addProductToWishlist = async ({
+  state,
+  dispatch,
+  setMessage,
+  setDisableButton,
+  product,
+  isRendered,
+  userId
+}) => {
+  !isAlreadyAdded(state.itemsInWishlist, product._id)
+    ? setMessage({ msg: "adding to wishlist..", msgType: "toast-inform" })
+    : setMessage({
+        msg: "removing from wishlist..",
+        msgType: "toast-inform"
+      });
+  setDisableButton(true);
+
+  try {
+    const {
+      data: { response },
+      status
+    } = await axios({
+      method: "POST",
+      url: `https://uandistoreapi.herokuapp.com/wishlists/${userId}/wishlist`,
+      data: {
+        _id: product._id
+      }
+    });
+
+    if (status === 200 || status === 201) {
+      dispatch({
+        type: "SET_WISHLIST",
+        payload: response
+      });
+    }
+  } catch (error) {
+    console.error(error);
```

Comment:
remove console

Reply:
Sure, I will remove all the console statements once I implement toast functionality, before deploying to production.